resources/RustDsc/RustDsc.psm1

```diff
@@ -0,0 +1,267 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+using namespace System.Collections.Generic
+
+#region Functions
+function Assert-Cargo {
+    # Refresh session $path value before invoking 'cargo'
+    $env:Path = [System.Environment]::GetEnvironmentVariable('Path', 'Machine') + ';' + [System.Environment]::GetEnvironmentVariable('Path', 'User')
+    try {
+        $null = Invoke-Cargo -Command 'help'
+    } catch {
+        throw 'Rust/Cargo is not installed'
+    }
+}
+
+function Invoke-Cargo {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$Command
+    )
+
+    return Invoke-Expression -Command "cargo $Command"
+}
+
+function Get-InstalledCargoCrates {
+    $result = Invoke-Cargo -Command 'install --list'
+    return $result
+}
+
+function Install-CargoCrate {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$CrateName,
+
+        [Parameter()]
+        [string]$Version,
+
+        [Parameter()]
+        [string[]]$Features,
+
+        [Parameter()]
+        [bool]$Force = $false
+    )
+
+    $command = [List[string]]::new()
+
+    $command.Add('install')
+    $command.Add($CrateName)
+    
+    if (-not([string]::IsNullOrEmpty($Version))) {
+        $command.Add('--version')
+        $command.Add($Version)
+    }
+
+    # Handle features
+    if ($null -ne $Features -and $Features.Count -gt 0) {
+        $command.Add('--features')
+        $command.Add(($Features -join ','))
+    } else {
+        # If no specific features are provided, assume all features
+        $command.Add('--all-features')
+    }
+
+    # Handle force flag
+    if ($Force) {
+        $command.Add('--force')
+    }
+
+    $command.Add('--quiet')
+
+    Write-Verbose -Message "Executing 'cargo $($command -join ' ')'"
+
+    return Invoke-Cargo -Command ($command -join ' ')
```

Comment:
I am not familiar with cargo, but I am curious if any of the arguments have spaces, this would possibly cause unexpected behavior. Should we update the implementation of Invoke-Cargo to call `& cargo @command`. Something like this:

```ps
function Invoke-Cargo {
    param (
        [Parameter(Mandatory = $true)]
        [string[]]$Command
    )

    & cargo @Command
}
```